pkg/watcher/certificate_manager.go

```diff
@@ -279,35 +277,17 @@ func (e *CertificateNotReadyError) Error() string {
 	return "Certificate-Secret does not exist"
 }
 
-func (c *CertificateManager) GetCACertificateStatus(ctx context.Context) (certmanagerv1.CertificateStatus, error) {
-	cachedCertStatus := c.caCertCache.GetCACertStatusFromCache(c.config.CACertificateName)
-
-	if cachedCertStatus.RenewalTime == nil || certificateRenewalTimePassed(cachedCertStatus) {
-		caCert := certmanagerv1.Certificate{}
-		if err := c.kcpClient.Get(ctx,
-			client.ObjectKey{Namespace: c.config.IstioNamespace, Name: c.config.CACertificateName},
-			&caCert); err != nil {
-			return certmanagerv1.CertificateStatus{}, fmt.Errorf("failed to get CA certificate %w", err)
-		}
-		c.caCertCache.SetCACertToCache(caCert)
-		return caCert.Status, nil
-	}
-
-	return cachedCertStatus, nil
-}
-
 func (c *CertificateManager) RemoveSecretAfterCARotated(ctx context.Context, kymaObjKey client.ObjectKey) error {
-	caCertificateStatus, err := c.GetCACertificateStatus(ctx)
+	gatewaySecret, err := gatewaysecret.GetGatewaySecret(ctx, c.kcpClient)
```

Comment:
This object will be fetched very frequently. In the previous implementation we used an explicit cache. Is this call cached properly?

Reply:
all secret objects are [cached by the Kyma reconciler](https://github.com/kyma-project/lifecycle-manager/blob/39caf9bb6be863211bde44c17b392cd296bfabb2/internal/controller/kyma/setup.go#L67)